passagemath-ecl fails to build with `std=gnu23`

```
  [sagemath_ecl-10.8.3.rc2]   [spkg-install] gcc -std=gnu23 -g -O2 -Isage/libs -I/usr/local/opt/gmp/include -I/usr/local/Cellar/ecl/24.5.10/include -I/Users/runner/work/passagemath/passagemath/src/sage/cpython -I/private/var/folders/bf/5my00kcs76g2n6kgkmsmvfdc0000gn/T/build-env-guqaeupv/lib/python3.14/site-packages/cysignals -I/Users/runner/work/passagemath/passagemath/src -I/usr/local/opt/python@3.14/Frameworks/Python.framework/Versions/3.14/include/python3.14 -I/private/var/folders/bf/5my00kcs76g2n6kgkmsmvfdc0000gn/T/build-env-guqaeupv/include -I/usr/local/opt/python@3.14/Frameworks/Python.framework/Versions/3.14/include/python3.14 -c sage/libs/ecl.c -o build/temp.macosx-15.0-x86_64-cpython-314/sage/libs/ecl.o -Ddarwin
  [sagemath_ecl-10.8.3.rc2]   [spkg-install] In file included from sage/libs/ecl.c:1177:
  [sagemath_ecl-10.8.3.rc2]   [spkg-install] In file included from /usr/local/Cellar/ecl/24.5.10/include/ecl/ecl.h:81:
  [sagemath_ecl-10.8.3.rc2]   [spkg-install] /usr/local/Cellar/ecl/24.5.10/include/ecl/object.h:27:13: error: redeclaration of C++ built-in type 'bool'
  [sagemath_ecl-10.8.3.rc2]   [spkg-install]    27 | typedef int bool;
  [sagemath_ecl-10.8.3.rc2]   [spkg-install]       |             ^
  [sagemath_ecl-10.8.3.rc2]   [spkg-install] 1 error generated.
  [sagemath_ecl-10.8.3.rc2]   [spkg-install] error: command '/usr/bin/gcc' failed with exit code 1
```
https://github.com/passagemath/passagemath/actions/runs/23609961721/job/68861243133#step:10:3187

--- a/src/sage/libs/ecl.pyx
+++ b/src/sage/libs/ecl.pyx
@@ -1,4 +1,5 @@
 # sage_setup: distribution = sagemath-ecl
+# distutils: extra_compile_args = -std=gnu17
 """
 Library interface to Embeddable Common Lisp (ECL)
 """

